js/src/views/Dapps/dappsStore.js

```diff
@@ -277,6 +291,12 @@ export default class DappsStore extends EventEmitter {
 
       this.apps = this.apps
         .filter((app) => !app.id || !newAppsIds.includes(app.id))
+        .filter((app) => { // check local dapp changes
+          if (app.type === 'local' && _local && apps.indexOf(app) === -1) {
```

Comment:
This additional filter seems redundant, local apps that are not part of `apps` array will be removed by previous condition anyway `!newAppsIds.includes(app.id)` (unless local apps doesn't have `id` field?)